File: python/minimum_depth_of_binary_tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def minDepth(self, root):
        if not root: 
            return 0
        else:
            return self.minHeight(root)
        
    def minHeight(self, node):
        if not node:
            return 1 << 31 - 1
        if not node.left and not node.right:
            return 1
        return min(self.minHeight(node.left), self.minHeight(node.right)) + 1

File: leetcode/java/MinimumDepthofBinaryTree.java
/**
 * Created by Robbert on 1/23/15.
 */
public class MinimumDepthofBinaryTree {
    public int minDepth(TreeNode root) {
        if (root == null) {
            return 0;
        } else {
            return minHeight(root);
        }
    }

    private int minHeight(TreeNode root) {
        if (root == null) {
            return Integer.MAX_VALUE;
        }

        if (root.left == null && root.right == null) {
            return 1;
        }

        int left = minHeight(root.left);
        int right = minHeight(root.right);

        return Math.min(left, right) + 1;
    }

}
